src/main/java/jenkins/plugins/git/GitSCMFileSystem.java

```diff
@@ -342,28 +360,37 @@ public SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull
                 } catch (URISyntaxException ex) {
                     listener.getLogger().println("URI syntax exception for '" + remoteName + "' " + ex);
                 }
-                String prefix = Constants.R_HEADS; 
+                String prefix = Constants.R_HEADS;
                 if(branchSpec.getName().startsWith(Constants.R_TAGS)){
-                    prefix = Constants.R_TAGS; 
+                    prefix = Constants.R_TAGS;
                 }
                 String headName;
                 if (rev != null) {
                     headName = rev.getHead().getName();
                 } else {
                     if (branchSpec.getName().startsWith(prefix)){
-                        headName = branchSpec.getName().substring(prefix.length()); 
+                        headName = branchSpec.getName().substring(prefix.length());
                     } else if (branchSpec.getName().startsWith("*/")) {
                         headName = branchSpec.getName().substring(2);
                     } else {
                         headName = branchSpec.getName();
                     }
                 }
-                client.fetch_().prune(true).from(remoteURI, Arrays
-                        .asList(new RefSpec(
-                                "+" + prefix + headName + ":" + Constants.R_REMOTES + remoteName + "/"
-                                        + headName))).execute();
+                headName = expand(headName, env).trim();
```

Comment:
I discovered that this doesn't behave in quite the same way for statically provided strings vs environment variables in that by expanding the variable here rather than earlier we're forgoing some of the input normalization being done above. 

example:
`*/master` will become `refs/remotes/origin/master` (probably) whereas 

`${SOME_ENV_VAR}` whose value is `*/master` will become `refs/remotes/origin/*/master` which will NPE a little later on in the code. 

Obviously it's easy enough to just not include */ in the variable but it may also be worthwhile for variables and static strings to go through the same process of string normalization. 🤷

Reply:
`*/master` vs `master` does not make any difference when expansion is not used? I wonder why is it supported in non-expansion mode in this case. Perhaps keep expansion as implemented here and consider changing non-expansion version later?